src/legacy/btchip_apdu_hash_input_finalize_full.c

```diff
@@ -70,12 +71,17 @@ static bool check_output_displayable() {
     isOpCall =
         btchip_output_script_is_op_call(btchip_context_D.currentOutput + 8,
           sizeof(btchip_context_D.currentOutput) - 8);
-    if (((G_coin_config->kind == COIN_KIND_QTUM || G_coin_config->kind == COIN_KIND_HYDRA) &&
-         !btchip_output_script_is_regular(btchip_context_D.currentOutput + 8) &&
-         !isP2sh && !(nullAmount && isOpReturn) && !isOpCreate && !isOpCall) ||
-        (!(G_coin_config->kind == COIN_KIND_QTUM || G_coin_config->kind == COIN_KIND_HYDRA) &&
-         !btchip_output_script_is_regular(btchip_context_D.currentOutput + 8) &&
-         !isP2sh && !(nullAmount && isOpReturn))) {
+    if (btchip_output_script_is_regular(btchip_context_D.currentOutput + 8) ||
+        isP2sh || !(nullAmount && isOpReturn)) {
```

Comment:
This is incorrect. It would make any output script accepted if `!(nullAmount && isOpReturn)` is `true`, that is if `(nullAmount && isOpReturn)` is `false`, that is either the output is not null, or it is not an OP_RETURN. That is basically _any_ output script.

The negation should be removed.

Reply:
nice catch, all of negations had to be removed and I've accidentally left this one.